workspace/TS100/Core/Src/main.cpp

```diff
@@ -78,6 +78,7 @@ int main(void) {
 
 	resetWatchdog();
 
+	postRToSInit();
```

Comment:
By its name, this has to be started _after_ the scheduler for devices that use FreeRTOS calls such as vTaskDelay(). This will break on TS80

Reply:
Well, probably the name should be changed then. At the moment the function just creates the I2C semaphore and that's all, for both TS100 and TS80 AFAICT. vTaskDelay() is called by power_probe() from the MOV task, so the scheduler is guaranteed to be started by then.

Reply:
`post` meaning after I thought was fairly obvious 🤔 
And yeah nothing in there currently needs to be after the RTOS has started; but felt nicer to have a pre/post pair to do stuff that needs to be before / after kernel starts :|

Reply:
TBH, I'm not familiar with the concept of initialisations taking part after the scheduler is started, so ``post`` name wasn't immediately obvious to me. And I find it confusing that code in one task can depend on initialisations happening in another task, that feels fragile (in this specific case this task has higher priorities but in case someone is experimenting with the priorities or or selectively disables tasks, this can break).

Reply:
I'm not against another way of doing it; this was just fairly easy and low cost way to call it incase other types want to use osDelay during setup (for example if negotiations for a protocol are going to take longer may as well run the other threads more

Reply:
My understanding is that we have two different jobs to do:
1. Prepare IPC primitives (semaphores, mutexes, etc.)
2. Do some initialisation that can be done in parallel to the other tasks (such as QC negotiation).
Since the jobs are inherently different (and it's natural to assume the IPC should be ready before the tasks have any chance to use it) my proposal would be to:
1. Create another function to be called right before scheduler startup specifically for IPC init;
2. Move power_probe() inside postRTosInit().